File: pl/python/first_class_function4.py
# επιστροφή συνάρτησης από συνάρτηση

# η συνάρτηση logger επιστρέφει μια συνάρτηση
def logger(msg):
    def log_message():
        print("Log:", msg)

    return log_message


log_hi = logger("Hi!")
log_hi()

# Log: Hi!

File: pl/javascript/first_class_function4.js
// η συνάρτηση logger επιστρέφει μια συνάρτηση
function logger(msg) {
    function log_message(){
        console.log('Log: ' + msg)
    }
    return log_message;
}

log_hi = logger("Hi!")
log_hi()


// Log: Hi!